ServiceApp/src/org/droidplanner/services/android/core/drone/manager/MavLinkDroneManager.java

```diff
@@ -261,18 +262,36 @@ protected boolean executeAsyncAction(Action action, ICommandListener listener) {
         String type = action.getType();
         Bundle data = action.getData();
 
+        Timber.d("executeAsyncAction(): action=%s", type);
+
         switch (type) {
             //FOLLOW-ME ACTIONS
             case FollowMeActions.ACTION_ENABLE_FOLLOW_ME:
                 data.setClassLoader(FollowType.class.getClassLoader());
+
+                FollowLocationSource locationSource = data.getParcelable(FollowMeActions.EXTRA_LOCATION_SOURCE);
+
+                if(locationSource == null) {
+                    locationSource = FollowLocationSource.Internal;
+                }
+
                 FollowType followType = data.getParcelable(FollowMeActions.EXTRA_FOLLOW_TYPE);
-                enableFollowMe(followType, listener);
+                enableFollowMe(followType, locationSource, listener);
                 return true;
 
             case FollowMeActions.ACTION_UPDATE_FOLLOW_PARAMS:
                 if (followMe != null) {
                     data.setClassLoader(LatLong.class.getClassLoader());
 
+                    if(data.containsKey(FollowMeActions.EXTRA_LOCATION_SOURCE)) {
+                        FollowLocationSource source = data.getParcelable(FollowMeActions.EXTRA_LOCATION_SOURCE);
+                        if(source == null) {
+                            source = FollowLocationSource.Internal;
```

Comment:
@kellyschrock if the `source` is `null`, let's just not update the location source for the `followMe`. Doing that also allows up to skip the first `if` statement, and retrieve the value directly and check for non nullity.